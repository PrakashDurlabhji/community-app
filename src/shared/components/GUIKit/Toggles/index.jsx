/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
/**
 * Toggles component.
 */
import React, { useRef, useState } from 'react';
import PT from 'prop-types';
import _ from 'lodash';
import './style.scss';

import { config } from 'topcoder-react-utils';

function Toggles({ checked, onChange, size }) {
  const [internalChecked, setInternalChecked] = useState(checked);
  let sizeStyle = size === 'lg' ? 'lgSize' : null;
  const delayedOnChange = useRef(
    _.debounce((q, cb) => cb(q), config.GUIKIT.DEBOUNCE_ON_CHANGE_TIME),
  );

  if (!sizeStyle) {
    sizeStyle = size === 'xs' ? 'xsSize' : 'smSize';
  }

  return (
    <label styleName={`container ${sizeStyle}`}>
      <input
        checked={internalChecked}
        type="checkbox"
        onChange={(e) => {
          setInternalChecked(e.target.checked);
          delayedOnChange.current(e.target.checked, onChange);
        }}
      />
      <span styleName="slider" />
      {sizeStyle === 'lgSize' && !internalChecked ? <span styleName="off">off</span> : null}
      {sizeStyle === 'lgSize' && internalChecked ? <span styleName="on">on</span> : null}
    </label>
  );
}

Toggles.defaultProps = {
  checked: false,
  onChange: () => {},
  size: 'sm',
};

Toggles.propTypes = {
  checked: PT.bool,
  onChange: PT.func,
  size: PT.oneOf(['xs', 'sm', 'lg']),
};

export default Toggles;
