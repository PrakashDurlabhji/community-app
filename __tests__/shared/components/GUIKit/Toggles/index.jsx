import React from 'react';
import Renderer from 'react-test-renderer/shallow';

import Toggles from 'components/GUIKit/Toggles';


const rnd = new Renderer();

it('Default render', () => {
  rnd.render((<Toggles />));
  expect(rnd.getRenderOutput()).toMatchSnapshot();
});
