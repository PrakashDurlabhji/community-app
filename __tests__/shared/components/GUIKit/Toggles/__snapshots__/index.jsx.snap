// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default render 1`] = `
<label
  className="src-shared-components-GUIKit-Toggles-___style__container___Ny137 src-shared-components-GUIKit-Toggles-___style__smSize___3FNLG"
>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="src-shared-components-GUIKit-Toggles-___style__slider___A0xAb"
  />
</label>
`;
